### PR TITLE
Bug 1220584 - [TV] FTE for Casting video and website to TV

### PR DESCRIPTION
**Pull request by Fischer-L**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c96952a736abb1620a4493403d0c69e00a89faf2

File: build/config/tv/apps-engineering.list [hunk 1]
```diff
@@ -13,6 +13,7 @@ tv_apps/notification-receiver
 tv_apps/weather-widget
 tv_apps/remote-control
 tv_apps/remote-control-client
+tv_apps/send-video-to-tv
```
Comment (rexboy7) [inline build/config/tv/apps-engineering.list]: You need to add it to `apps-production.list` too.

File: tv_apps/send-video-to-tv/offline_content/index.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      html {
+        width: 100%;
+        height: 100%;
+        color: #fff;
+      }
+    </style>
+  </head>
+
+  <body>
+    TODO: To add the content displayed on offline
```
Comment (rexboy7) [inline tv_apps/send-video-to-tv/offline_content/index.html]: So where are those contents?

File: build/config/tv/apps-engineering.list (hunk 1 above)
Comment (rexboy7) [inline build/config/tv/apps-engineering.list]: you need to add it to `apps-production.list` too.

File: tv_apps/send-video-to-tv/offline_content/index.html (hunk 2 above)
Comment (rexboy7) [inline tv_apps/send-video-to-tv/offline_content/index.html]: I think at least the logic for determining online/offline is required.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=93828ee6e1699eb257b46f5ee2323917de7afbe4

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f08b0cc0b5d22f16dd1dfde71da9601b4c89fcf7

File: tv_apps/fling-tutorial/offline_content/js/index.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  var tutorial = new FTEWizard('castingTutorial');
+  tutorial.init({
+    container: tutorialContainer,
+    pageClass: 'slide',
+    buttonsClass: 'slide-button',
+    launchEveryTime: true,
+    propagateKeyEvent: true,
+    onfinish: function () {
+      // Rewite the location to trigger the mozbrowserlocationchange event
```
Comment (rexboy7) [inline tv_apps/fling-tutorial/offline_content/js/index.js]: typo: Rewrite.

File: tv_apps/fling-tutorial/offline_content/js/index.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      // to let the outer app know user finishes this tutorial
+      window.location.href = window.location.href + '#finished';
+    }.bind(tutorial)
+  });
+
+  exports.tutorial = tutorial;
```
Comment (rexboy7) [inline tv_apps/fling-tutorial/offline_content/js/index.js]: I did some investigation but still can't find the specific reason why the button is not focused. Maybe you can try to listen to it on some event like `load.

File: tv_apps/fling-tutorial/js/index.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    }
+  });
+
+  // The mozbrowserclose event would be trrigered when content inside iframe
+  // is crashed. When crashed, let's just leave this app.
+  mozIframe.addEventListener('mozbrowserclose', () => {
+    window.close();
+  });
+});
+
```
Comment (rexboy7) [inline tv_apps/fling-tutorial/js/index.js]: I guess you need to do something like "Focus the iframe".

File: .jshintignore [hunk 6]
```diff
@@ -80,6 +80,8 @@ tv_apps/smart-home/bower_components/**
 tv_apps/smart-settings/bower_components/**
 tv_apps/smart-system/bower_components/**
 tv_apps/notification/bower_components/**
+tv_apps/fling-tutorial/bower_components/**
+tv_apps/fling-tutorial/offline_content/bower_components/**
```
Comment (rexboy7) [inline .jshintignore]: 1.`bower_components` theoretically should  be installed from bower.json, and you didn't included them. If you don't quite know about bower components I would suggest making a brief study.

2.I don't think making offline_content also a bower component is a good idea. Please don't just put everything under `offline_content`.

Comment (rexboy7) [inline .jshintignore]: I would prefer more to put directories like
`js/offline_content`
`style/offline_content`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8bdf918368df077c9b2af2b9c9a6e049f9a9a812

File: tv_apps/fling-tutorial/js/index.js (hunk 5 above)
Comment (Fischer-L) [inline tv_apps/fling-tutorial/js/index.js]: Updates, thank you.

File: .jshintignore (hunk 6 above)
Comment (Fischer-L) [inline .jshintignore]: Put these files outside the offline_content folder, thank you

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=dd31f0e5a72efacd9e2526dfa8f5e69550e3168f

File: .jshintignore (hunk 6 above)
Comment (rexboy7) [inline .jshintignore]: So why you still keep this line?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8d73f8d781d5f6deecb017723b2a85d39a3b1f8c

**Pull request merged**